lib/ui/channel_buffers.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart.ui;
+
+/// A saved platform message for a channel with its callback.
+class _StoredMessage {
+  /// Default constructor, takes in a [ByteData] that represents the
+  /// payload of the message and a [PlatformMessageResponseCallback]
+  /// that represents the callback that will be called when the message
+  /// is handled.
+  _StoredMessage(this._data, this._callback);
+
+  final ByteData _data;
+  final PlatformMessageResponseCallback _callback;
+
+  /// Getter for _data field, represents the message's payload.
+  ByteData get data => _data;
+  /// Getter for the _callback field, to be called when the message is received.
+  PlatformMessageResponseCallback get callback => _callback;
+}
+
+/// A fixed-size circular queue.
+class _RingBuffer<T> {
+  final collection.ListQueue<T> _queue;
+  int _capacity;
+  Function(T) _dropItemCallback;
+
+  _RingBuffer(this._capacity)
+    : _queue = collection.ListQueue<T>(_capacity);
+
+  int get length => _queue.length;
+
+  int get capacity => _capacity;
+
+  bool get isEmpty => _queue.isEmpty;
+
```

Comment:
That seemed redundant since the docstring for _dropItemCallback says all of that.  The intent was to point them to that documentation and not repeat it.

I found examples elsewhere and they put the ivar next to the setter and getter and it gets collapsed in the generated documentation so I don't have to repeat it, so done

Reply:
Yup, this works.